src/conversion/convertEditorConfig.ts

```diff
@@ -0,0 +1,79 @@
+import { FileSystem } from "../adapters/fileSystem";
+import { SansDependencies } from "../binding";
+import {
+    findEditorConfiguration,
+    DEFAULT_VSCODE_SETTINGS_PATH,
+} from "../input/findEditorConfiguration";
+import { TSLintToESLintSettings, ResultWithStatus, ResultStatus } from "../types";
+import { formatOutput } from "../creation/formatting/formatOutput";
+
+export type ConvertEditorConfigDependencies = {
+    findEditorConfiguration: SansDependencies<typeof findEditorConfiguration>;
+    fileSystem: Pick<FileSystem, "writeFile">;
+};
+
+/**
+ * Root-level driver to convert an editor configuration.
+ *
+ * Will currently only convert the rules:
+ *  "editor.codeActionsOnSave" -> "source.fixAll.tslint"
+ *  into
+ *  "eslint.autoFixOnSave"
+ *
+ * TODO: devinmotion: Replacements should be moved into rules set (by editor).
+ */
+export const convertEditorConfig = async (
+    dependencies: ConvertEditorConfigDependencies,
+    settings: TSLintToESLintSettings,
+): Promise<ResultWithStatus> => {
+    // 1. Existing configurations are read
+    const originalEditorConfiguration = await dependencies.findEditorConfiguration(settings.editor);
+    if (originalEditorConfiguration instanceof Error) {
+        return {
+            errors: [originalEditorConfiguration],
+            status: ResultStatus.Failed,
+        };
+    }
+
+    // TODO: devinmotion: The following logic may be moved to editor conversion rules.
+    // 2. Get relevant section in editor configuration
+    const codeActionsOnSave = originalEditorConfiguration["editor.codeActionsOnSave"];
+
+    // 3. Split properties to replace and original ones
+    const {
+        "source.fixAll.tslint": originalSourceFixAllTsLint,
+        ...codeActionsOnSaveWithoutReplacedProperties
+    } = codeActionsOnSave;
+
+    // Nothing to convert
+    if (!originalSourceFixAllTsLint) {
+        return {
+            status: ResultStatus.Succeeded,
+        };
+    }
+
+    // 4. Rewrite editor configuration with replacements.
+    const newEditorConfiguration = {
+        ...originalEditorConfiguration,
+        "editor.codeActionsOnSave": codeActionsOnSaveWithoutReplacedProperties,
+        "eslint.autoFixOnSave": originalSourceFixAllTsLint,
+    };
+
+    const output = newEditorConfiguration;
+    const outputPath = settings.editor ? settings.editor : DEFAULT_VSCODE_SETTINGS_PATH;
```

Comment:
Mmhh, yeah, this looks like it should be moved to a different function that's passed as a part of `dependencies` to here. It'd be hard to test as-is here (reasonably so - this function is pretty big). I think that's what you meant by the comments?

Reply:
Will work on this. Should work the same as convertRules in convertConfig.

Reply:
@JoshuaKGoldberg I splitted readingConfig, conversion (with extensible settings convertions), and writing of the results like it is done for the rules. I don't report any conversions yet. Tests have so be written. Could you please have a look at my new approach?

Reply:
Flagging this to look at it soon!

Reply:
Sorry for the delay @devinmotion! i wanted to give this a good reading-through.

Yes, this looks like a great approach. I think it'll work quite well, and it looks really polished already. Awesome! 🚀 

A few notes:

* `reportSettingConversionResults` seems like a subset of `reportConversionResults`. I wonder if you could have a single function that receives a standardized input object and logs the output in a similar way between them? Seems like you'd remove a good portion of copypasta from that particular new file.
* Similarly, `convertConfig` and `convertEditorConfig` are _really_ similar... I wonder if the common parts from both should be moved to some common `runFileConversion` helper? 🤷‍♂ maybe better to file a followup ticket to investigate; that feels like a kind of long-winded architectural debate.
* Reading through the branch, the difference between general linter config names (e.g. `convertConfig`) and their settings file equivalents (e.g. `convertSettingsConfig`) was a little hard to keep track of. That seems like something good for a followup issue+PR as well to clarify though, to keep the PR from getting too massive...

Reply:
No problem. I really appreciate your extensive feedback. It is really helpful for me. 🙏 

As soon as I will find some time, I'll do the following to get the PR ready:
- Refactor `reportSettingConversionResults` and `reportConversionResults` to keep it more DRY
- Try to refactor `convertConfig` and `convertEditorConfig` to be more DRY. If this looks too costly, I'll rather create a followup issue
- Create a followup issue to improve on the `convertConfig` and `convertSettingsConfig` similarities and differences
- Write tests for new features

Reply:
Refactor reportSettingConversionResults and reportConversionResults to keep it more DRY ✔️ including tests

Reply:
Try to refactor convertConfig and convertEditorConfig to be more DRY. If this looks too costly, I'll rather create a followup issue ℹ️

Tried to do some refactorings but could not get that far because it seems to be complex without further changes of how rules and editor settings are converted.
Therefore I created a follow-up issue: https://github.com/typescript-eslint/tslint-to-eslint-config/issues/274

Reply:
Create a followup issue to improve on the convertConfig and convertSettingsConfig similarities and differences ✔️ 
Issue created, needs clarification first: https://github.com/typescript-eslint/tslint-to-eslint-config/issues/275

Reply:
🚧 Working on the last missing tests:
- `convertSettings.ts`
- `writeEditorConfigConversionResults.ts`
- `convertEditorConfig.ts`